mp/src/game/shared/neo/neo_gamerules.cpp

```diff
@@ -1828,24 +1867,86 @@ static CNEO_Player* FetchAssists(CNEO_Player* attacker, CNEO_Player* victim)
 	return NULL;
 }
 
+#ifdef GAME_DLL
+void CNEORules::CheckIfCapPrevent(CNEO_Player *capPreventerPlayer)
+{
+	// If this is the only player alive left before the suicide/disconnect and the other team was holding
+	// the ghost, reward the other team an XP to the next rank as a ghost cap was prevented.
+	if (neo_sv_suicide_prevent_cap_punish.GetBool() && m_nRoundStatus == NeoRoundStatus::RoundLive &&
+			!m_bTeamBeenAwardedDueToCapPrevent)
+	{
+		bool bOtherTeamPlayingGhost = false;
+		int iTallyAlive[TEAM__TOTAL] = {};
+		const int iPreventerTeam = capPreventerPlayer->GetTeamNumber();
+		// Sanity check: Make sure it's only Jinrai/NSF players
+		const bool bValidTeam = iPreventerTeam == TEAM_JINRAI || iPreventerTeam == TEAM_NSF;
+		Assert(bValidTeam);
+		if (bValidTeam)
+		{
+			const int iCapPreventerEntIdx = capPreventerPlayer->entindex();
+
+			// Sanity check: Prevent duplication just in-case
+			bool bContainsEntIdx = false;
+			for (int i = 0; i < m_iEntPrevCapSize; ++i)
+			{
+				bContainsEntIdx = (m_arrayiEntPrevCap[i] == iCapPreventerEntIdx);
+				if (bContainsEntIdx) break;
+			}
+			if (!bContainsEntIdx) m_arrayiEntPrevCap[m_iEntPrevCapSize++] = iCapPreventerEntIdx;
+
+			for (int i = 1; i <= gpGlobals->maxClients; ++i)
+			{
+				auto *player = static_cast<CNEO_Player*>(UTIL_PlayerByIndex(i));
+				if (!player || player->entindex() == iCapPreventerEntIdx)
+				{
+					continue;
+				}
+
+				const int iPlayerTeam = player->GetTeamNumber();
+				iTallyAlive[iPlayerTeam] += player->IsAlive();
+				if (iPlayerTeam != iPreventerTeam && player->IsCarryingGhost())
+				{
+					bOtherTeamPlayingGhost = true;
+				}
```

Comment:
Nit: could break early here

Reply:
The loops is also for tallying up all alive players from both teams so I wouldn't break early here.